### DISCUSSION TITLE
[New Feature] Priority Queue.

### DISCUSSION THREAD

**Discussion by Lylaktech** [Vinifera-Developers/Vinifera#1469] [Feature Requests]
### Description:
By pressing a hotkey and left clicking a unit, that unit will start getting trained immediately after current unit is finished, bypassing the rest of the units in the queue.

For example: Say you had a queue of 20 tanks and just got tech to build the mammoth MK2 and you want to start building it right now without cancelling the queue, you would priority queue it so that it starts training as soon as the current unit finishes.

**Comment by OmegaBolt** [comment]
I feel with this (and it would be helpful with existing queuing) there should be some identifier too. Like perhaps the text colour for 1st and 2nd+ in the queue. eg the 1st in the queue is yellow, anything later is white as normal. This would let you see which is the next item to build.

**Comment by Onidotmoe** [comment]
Maybe by Alt-Clicking on the cameo it will move it forward in the queue and Alt-Right-Clicking will move it backwards?

Having some indicator for queue order would also be super neat, like maybe having the position of the ordered thing displayed in the top right corner of the cameo.
